components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -815,7 +830,93 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
+                                                              String consentPageUrl, String sessionDataKeyConsent) {
+
+        if (isAuthEndpointRedirectParamsFilterConfigAvailable()) {
+            return FrameworkUtils.getRedirectURLWithFilteredParams(consentPageUrl,
+                    endpointParams);
+        } else {
+            return EndpointUtil.getRedirectURLWithFilteredParams(consentPageUrl,
+                    endpointParams, sessionDataKeyConsent);
+        }
+    }
+
+    private static String getConsentRequiredScopesAsString(Set<String> consentRequiredScopesSet) {
+
+        String consentRequiredScopes = StringUtils.EMPTY;
+        if (CollectionUtils.isNotEmpty(consentRequiredScopesSet)) {
+            consentRequiredScopes = String.join(" ", consentRequiredScopesSet).trim();
+        }
+        return consentRequiredScopes;
+    }
+
+    private static String getQueryString(OAuth2Parameters params, SessionDataCacheEntry entry) throws
+            UnsupportedEncodingException, OAuthSystemException {
+
+        String queryString;
+        queryString = entry.getQueryString();
+        if (queryString.contains(REQUEST_URI) && params != null) {
+            // When request_uri requests come without redirect_uri, we need to append it to the SPQueryParams
+            // to be used in storing consent data
+            queryString = queryString +
+                    "&" + PROP_REDIRECT_URI + "=" + URLEncoder.encode(params.getRedirectURI(), UTF_8);
+        }
+
+        if (params != null) {
+            queryString = queryString + "&" + PROP_OIDC_SCOPE +
+                    "=" + URLEncoder.encode(StringUtils.join(getRequestedOIDCScopes(params), " "), UTF_8);
+        }
+        entry.setQueryString(queryString);
+        queryString = URLEncoder.encode(queryString, UTF_8);
+        return queryString;
+    }
+
+    private static boolean isAuthEndpointRedirectParamsFilterConfigAvailable() {
+        return FileBasedConfigurationBuilder.getInstance().isAuthEndpointRedirectParamsConfigAvailable();
+    }
+
+    /**
+     * Returns the consent page URL after filtering the query params.
+     *
+     * @param redirectUrl           The redirect URL.
+     * @param endpointParams        The map for store filtered params.
+     * @param sessionDataKeyConsent The value of sessionDataKeyConsent.
+     * @return redirect URL after filtering the query params except the sessionDataKeyConsent.
+     */
+    private static String getRedirectURLWithFilteredParams(String redirectUrl,
+                                                          Map<String, Serializable> endpointParams, String
+                                                                   sessionDataKeyConsent) {
+
+        URIBuilder uriBuilder;
+
+        try {
+            uriBuilder = new URIBuilder(redirectUrl);
+        } catch (URISyntaxException e) {
+            log.warn("Unable to filter redirect params for url." + redirectUrl, e);
```

Comment:
Shouldn't we break the flow here?
Otherwise we might expose sensitive data in the URL?

Reply:
Resolved by https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2020/commits/34e5840aee4cd6fdbff4d02bbb790cd6bc556127
Now the flow will break when unable to filter redirect prams from the url.